### ISSUE TITLE
Relative paths in Cargo.toml readme

### ISSUE THREAD

**Issue by kornelski** [rust-lang/cargo#5911]
I've found an odd case:

https://github.com/graphql-rust/juniper/blob/56f71e934be814722d10b0660156cf56b3753470/juniper/Cargo.toml#L12

1. The `readme` path is relative pointing to a parent directory: `../README.md`,
2. there is no `README.md` in the `.crate` file,
3. and yet, miraculously, crates.io shows a readme: https://crates.io/crates/juniper/0.9.2

It seems the readme was snatched when the crate was published, with whatever filesystem access was temporarily possible at time of publishing. Unfortunately, it means it's not possible to get the README later from the package.

Could Cargo require the README to be in the package?

**Comment by dwijnand** [comment]
What should be done for this case?  It's just a bit of an odd one.

**Comment by kornelski** [comment]
1. When publishing, validate the `readme` path and check that it doesn't escape crate's root (`root.join(readme).strip_prefix(root).is_some()` perhaps?)
2. Check that the path is included in the package. If the crate uses `include` key, perhaps add it there, or warn if it's excluded.

**Comment by dwijnand** [comment]
Yeah, there's a few different ways cargo could be more consistent, the whole spectrum from hard erroring, to warning, to accommodating.  I'm just not sure what we prefer. 🙂

**Comment by dwijnand** [comment]
Mildly related, on publishing and README checking: #4861

**Comment by dwijnand** [comment]
In #4861 @ehuss suggest this fix:

> copy the README somewhere into the package and rewrite the path in Cargo.toml

**Comment by DarkEld3r** [comment]
I have just run into this issue. I think that it would be great if cargo could deal with it copying files outside of the crate directory, but in the meantime it would be nice to have a warning (personally I would prefer an error).

**Comment by jmyersmsft** [comment]
Perhaps #7905 could provide inspiration and/or implementation for a fix here

**Comment by rodrigocfd** [comment]
I had this same exact question, and apparently it can be solved by adding the relative path to the `include` array:

```toml
[package]
readme = "../README.md"
include = ["../README.md"]
```

**Comment by epage** [comment]
It looks like this was fixed in 3bbb781
